### ISSUE TITLE
doc(readme): add technology icons and cover image

### ISSUE THREAD

**Issue by MFarabi619** [cuweb/rds#412]
## **What is the problem and why does it matter?**

The current GitHub README for the [Raven Design System](https://github.com/cuweb/rds) lacks visual appeal and could be more engaging.

![image](https://github.com/user-attachments/assets/6f64f6b1-7b11-4edf-9c17-7b923a35adbc)

To address this, a proposed cover design has been created in the following [draft Figma file](https://www.figma.com/design/ePJyecoY7yfS5YYHSRB5QP/Raven-Design-System?node-id=0-1&t=sowOhUvLjBUqqyQm-1).

![readme_cover](https://github.com/user-attachments/assets/be24b2bf-de3a-4e95-bb77-6a376e296215)

A Tokens Studio configuration has also been initialized to document design tokens and keep them in sync.

![image](https://github.com/user-attachments/assets/eb7908f3-f39e-45bb-9c1d-63976aad2080)

Additionally, technology icons can be sourced from platforms like [devicon.dev](https://devicon.dev) for further enhancement.

The re-structured section will appear as follows:

## Technologies Used

| [TypeScript](https://www.typescriptlang.org/) | [Vite](https://vitejs.dev/) | [Cypress](https://www.cypress.io/) | [Storybook](https://storybook.js.org/) | [React](https://reactjs.org/) | [Tailwind CSS](https://tailwindcss.com/) |
|:---------------------------------------------:|:----------------------------:|:----------------------------------:|:--------------------------------------:|:-----------------------------:|:---------------------------------------:|
| [![TypeScript](https://cdn.jsdelivr.net/gh/devicons/devicon@latest/icons/typescript/typescript-original.svg)](https://www.typescriptlang.org/) | [![Vite](https://cdn.jsdelivr.net/gh/devicons/devicon@latest/icons/vitejs/vitejs-original.svg)](https://vitejs.dev/) | [![Cypress](https://cdn.jsdelivr.net/gh/devicons/devicon@latest/icons/cypressio/cypressio-original.svg)](https://www.cypress.io/) | [![Storybook](https://cdn.jsdelivr.net/gh/devicons/devicon@latest/icons/storybook/storybook-original.svg)](https://storybook.js.org/) | [![React](https://cdn.jsdelivr.net/gh/devicons/devicon@latest/icons/react/react-original.svg)](https://reactjs.org/) | [![Tailwind CSS](https://cdn.jsdelivr.net/gh/devicons/devicon@latest/icons/tailwindcss/tailwindcss-original.svg)](https://tailwindcss.com/) |

The re-ordering follows the categorization of programming langauges, build tools, testing tools, and front-end libraries.

Adding badges for things like conventional commits and commitizen would also be a nice touch.

## **What is the proposed solution?**

Refresh the GitHub README by incorporating the proposed design, making it more visually appealing and informative. This update will improve first impressions and better showcase the project. The final README will look as follows:

<h1 align="center">Raven Design System</h1>

![readme_cover](https://github.com/user-attachments/assets/be24b2bf-de3a-4e95-bb77-6a376e296215)

[![npm](https://img.shields.io/npm/v/@carletonuniversity/rds)](https://www.npmjs.com/package/@carletonuniversity/rds) [![Conventional Commits](https://img.shields.io/badge/Conventional%20Commits-1.0.0-%23FE5196?logo=conventionalcommits&logoColor=white)](https://conventionalcommits.org) [![Commitizen friendly](https://img.shields.io/badge/commitizen-friendly-brightgreen.svg)](http://commitizen.github.io/cz-cli/) [![Storybook](https://raw.githubusercontent.com/storybooks/brand/master/badge/badge-storybook.svg)](https://github.com/storybooks/storybook) ![Repo Size](https://img.shields.io/github/repo-size/cuweb/rds)

Raven Design System, better known as RDS, is Carleton University’s design system, is a living digital organism uniting campus wide teams around a common visual language.

## Technologies Used

| [TypeScript](https://www.typescriptlang.org/) | [Vite](https://vitejs.dev/) | [Cypress](https://www.cypress.io/) | [Storybook](https://storybook.js.org/) | [React](https://reactjs.org/) | [Tailwind CSS](https://tailwindcss.com/) |
|:---------------------------------------------:|:----------------------------:|:----------------------------------:|:--------------------------------------:|:-----------------------------:|:---------------------------------------:|
| [![TypeScript](https://cdn.jsdelivr.net/gh/devicons/devicon@latest/icons/typescript/typescript-original.svg)](https://www.typescriptlang.org/) | [![Vite](https://cdn.jsdelivr.net/gh/devicons/devicon@latest/icons/vitejs/vitejs-original.svg)](https://vitejs.dev/) | [![Cypress](https://cdn.jsdelivr.net/gh/devicons/devicon@latest/icons/cypressio/cypressio-original.svg)](https://www.cypress.io/) | [![Storybook](https://cdn.jsdelivr.net/gh/devicons/devicon@latest/icons/storybook/storybook-original.svg)](https://storybook.js.org/) | [![React](https://cdn.jsdelivr.net/gh/devicons/devicon@latest/icons/react/react-original.svg)](https://reactjs.org/) | [![Tailwind CSS](https://cdn.jsdelivr.net/gh/devicons/devicon@latest/icons/tailwindcss/tailwindcss-original.svg)](https://tailwindcss.com/) |

## Getting Started

1. In a terminal window open your install location and run the following:

   ```shell
   git clone https://github.com/cuweb/rds.git
   ```

2. Once the clone is complete move into the `RDS` directory and run:

   ```shell
   npm install
   ```

3. Run Storybook

   ```shell
   npm run storybook
   ```

4. Other useful shell commands

- `npm run rebuild` - deletes node modules, package lock and run npm install
- `npm run format` - format files with prettier
- `npm run lint` - check for errors with eslint
- `npm run size` - compare compiled size against pre-defined limit

## Component Testing

Run Storybook and Cypress concurrently

```
npm run test
```

You can also run Storybook and Cypress individually by using the following commands in different terminal windows:

- Run Storybook - `npm run storybook`
- Run Cypress - `npx cypress Open`

### Other Commands

- Run tests in terminal while storybook is being launched - `npm run test:ci`

## Comment Linting

When adding a commit with a commit you may be blocked if you do not prefix your comment with one of the following types. So a commit should look like: `git commit -m "change: updated spacing around button"`. The prefix requires a colon to separate it from the comment.

- `add` - used when adding something new such as a component, story or test file
- `change` - used when making a change to an existing file in the project
- `deprecate` - used when deprecating code
- `doc` - used when updating something related to documentation
- `fix` - used when fixing an error, warning or something else
- `finish` - used when finishing a feature or fix
- `refactor` - used when refactoring something such as a component or build tool
- `release` - used when preparing a new release
- `revert` - used when revert to a past commit
- `test` - used when making changes relating to testing
- `upgrade` - used when upgrading packages

## Deployment

This project is available as an [NPM Package](https://www.npmjs.com/package/@carletonuniversity/rds).

## How to use in a project

1. Install the package using one of the following command:

   ```shell
   yarn add @carletonuniversity/rds
   npm install @carletonuniversity/rds
   ```

2. Import the css file into your primary css file using:

   ```css
   /* primarycssfile.css */
   @import '../node_modules/@carletonuniversity/rds/dist/index.css';
   ```

   OR

   ```css
   /* primarycssfile.css */
   @import '~@carletonuniversity/rds/dist/index.css';
   ```

**Comment by troychaplin** [comment]
Thank you for the suggestion. There are improvements to be made in several areas of documentation that we hope to add as we move close to a full release of version 1.0 or RDS

**Comment by MFarabi619** [comment]
Closed by #413
